### ISSUE TITLE
.navbar-text or .form-control-static  in navbar forms

### ISSUE THREAD

**Issue by CriseX** [twbs/bootstrap#16266]
I find that neither really works well, while the static control displays nice it has the incorrect color the .navbar-text messes up the height of the navbar obviously.

Currently I fix this with .navbar-right on the form element and the text outside of the form itself, with .navbar-text, but it is not really a one size fits all solution.

**Comment by cvrebert** [comment]
Please post a JS Bin example of the problem.

**Comment by CriseX** [comment]
http://jsbin.com/webezibiyo/1/edit?html,output

There, just fiddle with the element visibility, like I said neither works out of the box as one would expect intuitively. Like I said moving the text outside the form does work, but for that to be the case the form has to be floated which makes that not a good solution for all use cases (although it is fine with mine, bar the odd nested float to get the inline display effect on the form element).

**Comment by twbs-lmvtfy** [comment]
Hi @CriseX!

You appear to have posted a live example (http://jsbin.com/webezibiyo/1/edit), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* [W002](https://github.com/twbs/bootlint/wiki/W002): `<head>` is missing X-UA-Compatible `<meta>` tag that disables old IE compatibility modes
* [W003](https://github.com/twbs/bootlint/wiki/W003): `<head>` is missing viewport `<meta>` tag that enables responsiveness
* line 5, column 1: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.4 ; saw what appears to be usage of Bootstrap 3.3.2
* line 6, column 1: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.4 ; saw what appears to be usage of Bootstrap 3.3.2

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by CriseX** [comment]
Sigh, again though I am sure I picked the latest version when creating that bin... oh well let us try again.

http://jsbin.com/qekutoweyu/1/edit?html,output

**Comment by CriseX** [comment]
Just a short addendum, being able to use a static control inside navbar-form out of the box would result in nicer display when the navbar is collapsed also vs what you get. with navbar-text outside the form itself.

**Comment by cvrebert** [comment]
I believe `.navbar-text` is only supposed to be used as a child of `.navbar-collapse`.
What color `.form-control-static` ought to be here seems like a design decision, which I'm not qualified to make.
CC: @mdo for confirmation of the former and a design decision on the latter.

**Comment by CriseX** [comment]
Just to point out, the same question about color can be raised about labels in navbar (not a typical use case, but it  dawned on me that semantically mine is more of a label than a static form value at least, although it doesn't really have an associated control as it is more of a leading text).

So semantically speaking moving it outside the form entirely and using navbar-text would be the correct choice but a) that results in the float trickery being necessary (or display: inline, I assume) for the form element and b) we get wrong dividers when navbar is collapsed, so we loose the visual attachment between the leading text and the form button.

If the navbar supported split dropdown nav links inside navbar-nav this issue wouldn't exist, because the use case of using a button like I do would not be as common either, unless you want the specific look and color coding provided by buttons (but let's pretend I did not just say that because it would turn this into an out of scope request, for version 3 at least).

**Comment by mdo** [comment]
The color could be extended to the `.form-control-static`, but we can't blindly make that decision as we don't know where in the DOM those classes will be used and against what background. Might be a won't fix for v3.

**Comment by mdo** [comment]
Won't fix it is.
